packages/wrangler/src/inspect.ts

```diff
@@ -0,0 +1,526 @@
+    setRetryRemoteWebSocketConnectionSigil((x) => x + 1);
+  }
+
+  /** A simple incrementing id to attach to messages we send to devtools */
+  const messageCounterRef = useRef(1);
+
+  /**
+   * We want to make sure we don't lose any messages we receive from the
+   * remote websocket before devtools connects. So we use a ref to buffer
+   * messages, and flush them whenever devtools connects.
+   */
+  const messageBufferRef = useRef<string[]>([]);
+
+  // This effect tracks the connection to the remote websocket
+  // (stored in, no surprises here, `remoteWebSocket`)
+  useEffect(() => {
+    if (!props.inspectorUrl) {
+      return;
+    }
+    // The actual websocket instance
+    const ws = new WebSocket(props.inspectorUrl);
+    setRemoteWebSocket(ws);
+
+    /**
+     * A handle to the interval we run to keep the websocket alive
+     */
+    let keepAliveInterval: NodeJS.Timer;
+
+    /**
+     * Test if the websocket is closed
+     */
+    function isClosed() {
+      return (
+        ws.readyState === WebSocket.CLOSED ||
+        ws.readyState === WebSocket.CLOSING
+      );
+    }
+
+    /**
+     * Send a message to the remote websocket
+     */
+    function send(event: Record<string, unknown>): void {
+      if (!isClosed()) {
+        ws.send(JSON.stringify(event));
+      }
+    }
+
+    /**
+     * Closes the inspector.
+     */
+    function close(): void {
+      if (!isClosed()) {
+        try {
+          ws.close();
+        } catch (err) {
+          // Closing before the websocket is ready will throw an error.
+        }
+      }
+    }
+
+    /**
+     * Since we have a handle on the remote websocket, we can tap
+     * into its events, and log any pertinent ones directly to
+     * the terminal (which means you have insight into your worker
+     * without having to open the devtools).
+     */
+    if (props.logToTerminal) {
+      ws.addEventListener("message", (event: MessageEvent) => {
+        if (typeof event.data === "string") {
+          const evt = JSON.parse(event.data);
+          if (evt.method === "Runtime.exceptionThrown") {
+            const params = evt.params as Protocol.Runtime.ExceptionThrownEvent;
+            console.error(
+              "🚨", // cheesy, but it works
+              // maybe we could use color here too.
+              params.exceptionDetails.text,
+              params.exceptionDetails.exception.description
+            );
+          }
+          if (evt.method === "Runtime.consoleAPICalled") {
+            logConsoleMessage(
+              evt.params as Protocol.Runtime.ConsoleAPICalledEvent
+            );
+          }
+        } else {
+          // We should never get here, but who know is 2022...
+          console.error("unrecognised devtools event:", event);
+        }
+      });
+    }
+
+    ws.addEventListener("open", () => {
+      send({ method: "Runtime.enable", id: messageCounterRef.current });
+      // TODO: This doesn't actually work. Must fix.
```

Comment:
What's broken about it?

Reply:
Just doesn't enable the Network panel in exisiting devtools, works on our forked version.